gips/00XX-indexing-payments.md

```diff
@@ -0,0 +1,154 @@
+  - Gateways monitor the QoS per open agreement and assess if they need to cancel (on-chain) and/or replace any of them.
+
+This solution aims to:
+
+- compensate indexers for the exact amount of work they do.
+- allow indexers to gracefully exit the agreement if it can't be fully fulfilled.
+- cap the amount payers will spend on indexing. Though they might spend but get no service back if the cap is too low.
+- incentivize sufficient QoS by awarding agreements to indexers that provide it.
+
+[TODO]: # "probably missing more desired properties"
+
+### Indexing Indexer Selection Algorithm (IISA)
+
+As mentioned above, gateways get to choose who they work with. As a way to do this we propose an IISA is used. It considers Indexer Quality-of-Service, available stake (with delegation), and other parameters to find the optimal balance between QoS, load-balancing, economic security and price.
+
+[TODO]: # "detail the IISA?"
+
+### Arbitration and disputes
+
+Since collections are tied to "amount of work done" and this is reported by the indexers themselves, there needs to be a mechanism to punish malicious reporting. We propose here to use slashing. For this to be fair we need the "amount of work done" to be verifiable by an arbitrator in the case of a dispute.
+
+### Amount of work done
+
+As an initial iteration of the "amount of work done" proxy we propose to use indexed blocks and subgraph entities. In particular the amount to be collected by indexers shall be equal to:
+
+- A price per amount of blocks indexed (with a different price for each chain). These are charged incrementally, i.e. every time a POI is posted, the indexer collects a payment for the new blocks indexed since the last POI.
+- A price per amount of entities stored by the subgraph (same price for each chain). These are charged on a recurring basis per unit of time, for the total number of entities on the subgraph at the time the POI is posted times the time since the last POI.
+
+This approach is vulnerable to some forms of abuse and may also lead to denial-of-service scenarios (e.g. malicious users deploying subgraphs that are super expensive to index but get a very small entities count).
+
+We anticipate needing to perform research into alternative approaches in the near future. We theorize that "amount of work done" may be modeled as a **subgraph gas** unit down the line, similar to ethereum gas.
+
+### Minimum Viable Product
+
+While this proposal mostly focuses on the full Graph Horizon backed implementation, we are also developing an MVP that does not require smart contracts.
+
+In this MVP, we will roll out a new gateway component (the “dipper”) that will be controlled with a CLI. E&N users (e.g. the BD team) can be onboarded to this CLI and use it to add subgraphs to the list of supported subgraphs. Once a subgraph is in the list, the dipper will take care of finding Indexers and setting up indexing agreements with them.
```

Comment:
rather than saying `E&N users (e.g. the BD team)` I would generalize this to "gateway operators", as the code will be open source and other people may run a gateway and a dipper